### PR TITLE
Bug in runtests

### PR DESCRIPTION
**Pull request by abuts**
This should resolve the bug and let runtests report failure if no tests were run

### CODE REVIEW

**Comment by abuts** [comment]
@nickbattam-tessella  The way component was added (or rather identified) was changed in previous PR.

File: _test/disabled_tests.md [hunk 1]
```diff
@@ -7,7 +7,7 @@
   - test_labprobe_nonmpi (no ticket)
 
 - test_job_dispatcher_parpool.m
-  - test_job_submittion (no ticket)
+  - test_job_submittion (Its not a disabled test, but a tester to debug job submission on a cluster)
```
Comment (hsautessella) [inline _test/disabled_tests.md:10]: If this isn't an actual test, we could probably remove it from here.

Comment (abuts) [reply _test/disabled_tests.md:10]: may be name can be changed
The code was exteamly useful for debugging jobs on cluster so I do not want to remember how to do it when time come to run jobs on SCARF

And it is a good place to find it

Comment (abuts) [reply _test/disabled_tests.md:10]: as the first thing you will be doing testing cluster, is check that the cluster config is correct

This is the test, which should be run on SCARF only. For everything else its substantial waste of time. 
When we have is_jenkins & is_scarf -- this test should run

File: _test/shared/matlab_xunit/xunit/TestSuite.m [hunk 2]
```diff
@@ -79,14 +79,17 @@
             if nargin < 2
                 monitor = CommandWindowTestRunDisplay();
             end
+            if nargin<3
+                num_tests_runed = 0;
```
Comment (nickbattam-tessella) [inline _test/shared/matlab_xunit/xunit/TestSuite.m:83]: `num_tests_run`

File: _test/shared/matlab_xunit/xunit/TestSuite.m [hunk 3]
```diff
@@ -79,14 +79,17 @@
...
+            end
             
             monitor.testComponentStarted(self);
             did_pass = true;
             
             self.setUp();
             
             for k = 1:numel(self.TestComponents)
-                this_component_passed = self.TestComponents{k}.run(monitor);
+                [this_component_passed,num_tests_runed] = self.TestComponents{k}.run(monitor,num_tests_runed);
```
Comment (nickbattam-tessella) [inline _test/shared/matlab_xunit/xunit/TestSuite.m:92]: how is `num_tests_runed` being updated by this method?

Comment (abuts) [reply _test/shared/matlab_xunit/xunit/TestSuite.m:92]: TestComponent, which is acutal executive, adds one to its input value. 
But if you ask how the tests are running in detals -- I do not fully understand it. Its pretty engenious code. This is why Matlab get it for its unit tests (initially it was this code, though Matlab had spoiled it later)

File: _test/shared/matlab_xunit/xunit/TestSuite.m (hunk 2 above)
Comment (abuts) [reply _test/shared/matlab_xunit/xunit/TestSuite.m]: would not spend time on this, but fine, changed and merging without test

**Pull request merged**